### PR TITLE
chore: use class based approach for map

### PR DESCRIPTION
**Pull request by yhl25**

### CODE REVIEW

File: examples/src/main/java/io/numaproj/numaflow/examples/function/map/evenodd/EvenOddFunction.java [hunk 1]
```diff
@@ -3,15 +3,22 @@
...
 
 import java.io.IOException;
 
+/**
+ * This is a simple User Defined Function example which receives a message,
+ * and attaches a key to the message based on the value, if the value is even
+ * the key will be set as "even" if the value is odd the key will be set as
+ * "odd"
+ */
+
```
Comment (KeranYang) [inline examples/src/main/java/io/numaproj/numaflow/examples/function/map/evenodd/EvenOddFunction.java:17]: 👍

File: src/test/java/io/numaproj/numaflow/function/FunctionServerTest.java [hunk 2]
```diff
@@ -38,6 +38,28 @@ public class FunctionServerTest {
...
+        public MessageT[] processMessage(String key, Datum datum) {
+            return new MessageT[]{MessageT.to(
+                    TEST_EVENT_TIME,
+                    key + PROCESSED_KEY_SUFFIX,
+                    (new String(datum.getValue())
+                            + PROCESSED_VALUE_SUFFIX).getBytes())};
+        }
+    }
+
     private final BiFunction<String, Datum, Message[]> testMapFn =
```
Comment (KeranYang) [inline src/test/java/io/numaproj/numaflow/function/FunctionServerTest.java:63]: Can we remove `testMapFn` and `testMapTFn`?

File: src/main/java/io/numaproj/numaflow/function/reduce/ShutdownActor.java [hunk 3]
```diff
@@ -59,4 +60,12 @@ private void completedSuccessfully(String eof) {
         // if all the actors completed successfully, we can stop the shutdown actor.
         getContext().getSystem().stop(getSelf());
     }
+
+    // if we see dead letters, we need to stop the execution and exit
+    // to make sure no messages are lost
+    private void handleDeadLetters(DeadLetter deadLetter) {
```
[the remark this answers is not in the document]
Comment (yhl25) [reply src/main/java/io/numaproj/numaflow/function/reduce/ShutdownActor.java:66]: https://github.com/numaproj/numaflow-java/blob/ece736ee436375f8d9db5905f07677b413c156a5/src/main/java/io/numaproj/numaflow/function/FunctionService.java#L156

[the remark this answers is not in the document]
Comment (yhl25) [reply src/main/java/io/numaproj/numaflow/function/reduce/ShutdownActor.java]: Done

**Review by KeranYang — APPROVED** [review]

**Pull request merged**